Formula/yubikey-agent.rb

```diff
@@ -0,0 +1,72 @@
+class YubikeyAgent < Formula
+  desc "Seamless ssh-agent for YubiKeys and other PIV tokens"
+  homepage "https://filippo.io/yubikey-agent"
+  url "https://github.com/FiloSottile/yubikey-agent/archive/v0.1.3.tar.gz"
+  sha256 "58c597551daf0c429d7ea63f53e72b464f8017f5d7f88965d4dae397ce2cb70a"
+  license "BSD-3-Clause"
+  head "https://filippo.io/yubikey-agent", using: :git
+
+  depends_on "go" => :build
+  depends_on "pinentry-mac"
+
+  def install
+    ENV["GOPATH"] = HOMEBREW_CACHE/"go_cache"
+    mkdir bin
+    system "go", "build", "-trimpath", "-ldflags", "-X main.Version=v#{version}",
+        "-o", bin, "filippo.io/yubikey-agent"
+    prefix.install_metafiles
+  end
+
+  def post_install
+    (var/"run").mkpath
+    (var/"log").mkpath
+  end
+
+  def caveats
+    <<~EOS
+      To set up a new YubiKey, run this command:
+        yubikey-agent -setup
+
+      To use this SSH agent, set this variable in your ~/.zshrc and/or ~/.bashrc:
+        export SSH_AUTH_SOCK="#{var}/run/yubikey-agent.sock"
+    EOS
+  end
+
+  plist_options manual: "yubikey-agent -l #{HOMEBREW_PREFIX}/var/run/yubikey-agent.sock"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>EnvironmentVariables</key>
+        <dict>
+          <key>PATH</key>
+          <string>/usr/bin:/bin:/usr/sbin:/sbin:#{Formula["pinentry-mac"].opt_bin}</string>
+        </dict>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_bin}/yubikey-agent</string>
+          <string>-l</string>
+          <string>#{var}/run/yubikey-agent.sock</string>
+        </array>
+        <key>RunAtLoad</key><true/>
+        <key>KeepAlive</key><true/>
+        <key>ProcessType</key>
+        <string>Background</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/yubikey-agent.log</string>
+        <key>StandardOutPath</key>
+        <string>#{var}/log/yubikey-agent.log</string>
+      </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/yubikey-agent", "-help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.